ml-algorithms/src/main/java/org/opensearch/ml/engine/Predictable.java

```diff
@@ -31,7 +34,13 @@ public interface Predictable {
      * @param mlInput input data
      * @return predicted results
      */
-    MLOutput predict(MLInput mlInput);
+    default MLOutput predict(MLInput mlInput) {
+        throw new IllegalStateException("Method is not implemented");
+    }
+
+    default void predict(MLInput mlInput, MLTask mlTask, ActionListener<MLTaskResponse> actionListener) {
+
```

Comment:
Add default action? 
```
actionListener.onFailure(new IllegalStateException("Method is not implemented")); 
```

Reply:
Added and changed method parameter to remove unused mlTask and renamed the method to avoid conflict with `public MLOutput predict(MLInput mlInput, MLModel model)`